cmake/mcuboot.cmake

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# This file includes extra build system logic that is enabled when
+# CONFIG_BOOTLOADER_MCUBOOT=y.
+#
+# It builds signed binaries using imgtool as a post-processing step
+# after zephyr/zephyr.elf is created in the build directory.
+#
+# Since this file is brought in via include(), we do the work in a
+# function to avoid polluting the top-level scope.
+
+function(zephyr_mcuboot_tasks)
+  set(keyfile "${CONFIG_MCUBOOT_SIGNATURE_KEY_FILE}")
+
+  # Check for misconfiguration.
+  if("${keyfile}" STREQUAL "")
+    # No signature key file, no signed binaries. No error, though:
+    # this is the documented behavior.
+    return()
+  elseif(NOT EXISTS "${keyfile}")
+    message(FATAL_ERROR "Can't sign images for MCUboot: CONFIG_MCUBOOT_SIGNATURE_KEY_FILE=\"${keyfile}\" does not exist.")
+  elseif(NOT WEST)
+    message(FATAL_ERROR "Can't sign images for MCUboot: west not found. To fix, install west and ensure it's on PATH.")
+  elseif(NOT (CONFIG_BUILD_OUTPUT_BIN OR CONFIG_BUILD_OUTPUT_HEX))
+    message(FATAL_ERROR "Can't sign images for MCUboot: Neither CONFIG_BUILD_OUTPUT_BIN nor CONFIG_BUILD_OUTPUT_HEX is enabled, so there's nothing to sign.")
+  endif()
+
+  # Find imgtool. Even though west is installed, imgtool might not be.
+  # The user may also have a custom manifest which doesn't include
+  # MCUboot.
+  #
+  # Therefore, go with an explicitly installed imgtool first, falling
+  # back on mcuboot/scripts/imgtool.py.
+  if(IMGTOOL)
+    set(imgtool_path "${IMGTOOL}")
+  else()
+    # We can't use ZEPHYR_MCUBOOT_MODULE_DIR because mcuboot is not a
+    # "real" zephyr module. It contains no CMakeLists.txt or Kconfig
+    # files. At time of writing, its zephyr/module.yml file was just for
+    # sanitycheck.
+    #
+    # Therefore, the build system doesn't regard it as a module, and
+    # doesn't define ZEPHYR_MCUBOOT_MODULE_DIR.
+    execute_process(COMMAND ${WEST} list -f "{posixpath}" mcuboot
```

Comment:
I know this will change now, but remember, as per: https://github.com/zephyrproject-rtos/zephyr/pull/26060, we should avoid calling `west` directly, but instead through same Python as rest of build system.